third_party/move/move-vm/types/src/value_serde.rs

```diff
@@ -172,6 +174,112 @@ pub fn serialized_size_allowing_delayed_values(
     })
 }
 
+/// Count number of types constant_serialized_size would visit, used for gas charging.
```

Comment:
@ziaptos structs can have constant serialized size?

Reply:
My comment was that "Count number of types `constant_serialized_size` would visit" is incorrect. Mainly because `constant_serializeed_size` returns early if it encounters non-constant seriazible field:
```
match cur {
                    Some(cur_value) => total = total.map(|v| v + cur_value),
                    None => {
                        total = None; // <--------------------------------------------------------
                        break;
                    },
                }
```

Maybe re-comment to: The upper bound of the number of types `constant_serialized_size` would visit... or something like that?

**Also, given that these two functions here are so specific for the current native implementation, I'd just move them to the same file where the native function is defined (I'd say they belong there a lot more than here!).**

Reply:
ah yes, I should specify here this is an upper limit.

I can move these to bcs.rs. We should see if serialized_size_allowing_delayed_values should go there too.

Reply:
why two functions - we need to charge for work before we perform any useful work. 

Or we can have one function , that charges gas as it goes, not sure if that would have efficiency consequences - @vgao1996  ?